rfcs/QueryLevelNullability.md

```diff
@@ -1,100 +1,158 @@
+  - [Apollo](https://github.com/apollographql/apollo-tooling#code-generation) has a code generator for Android (Kotlin)
+    and iOS (Swift) clients
+  - [The Guild](https://www.graphql-code-generator.com/) has a TypeScript code generator for web clients
 
-According to [best practice](https://graphql.org/learn/best-practices/#nullability) GraphQL is
-a "nullable by default" language meaning that all properties are allowed to be `null`.
-This is in contrast to the two modern languages used on mobile clients, Swift and Kotlin,
-which are both non-null by default. In Swift and Kotlin, unless developers otherwise
-specify it, properties cannot be `null`.
+  GraphQL codegen tools typically accept a schema and a set of documents as input, and output code in a language of
+  choice that represents the data returned by those operations.
 
-This mismatch creates some dissonance for developers who are currently forced to deal with
-problems that commonly surround nullablility. This makes large numbers of null fields tedious and
-time-consuming to work with.
+  For example, the Apollo iOS codegen tool generates Swift types to represent each query document, as well as model types
+  representing the data returned from those queries. Notably, a nullable field in the schema becomes an `Optional`
+  property on the generated Swift model type, represented by `?` following the type name.
 
-This poses a problem for the clients that use codegened models. For example Apollo codegen provides 
-Swift/Kotlin types to represent queries used in the app. Nullable fields in the schema are represented
-as nullable properties on the resulting type, represented by `?` following the type name:
-```graphql
-query GetBusinessName($encid: String!) {
-  business(encid: $encid) {
-    name
+  In the example below, the `Business` schema type has a nullable field called `name`.
+  ```graphql
+  # Schema
+  type Business {
+    # The unique identifier for the business (non-nullable)
+    id: String!
+  
+    # The name of the business (nullable)
+    name: String
   }
-}
-```
-```swift
-struct GetBusinessNameQuery.Data.Business {
-  let name: String?
-}
-```
-In many cases, the client should error if the business `name` is null. While the schema can have the nullable
-fields for valid reasons (such as federation), the client wants to decide if it accepts a null value for]
-the result to simplify the client-side logic.  
+
+  # Document
+  query GetBusinessName($id: String!) {
+    business(id: $id) {
+      name
+    }
+  }
+  ```
+  At build time, Apollo generates the following Swift code (note: the code has been shortened for clarity).
+  ```swift
+  struct GetBusinessNameQuery {
+    let id: String
+
+    struct Data {
+      let business: Business?
+
+      struct Business {
+        /// Since the `Business.name` schema field is nullable, the corresponding codegen Swift property is `Optional`
+        let name: String?
+      }
+    }
+  }
```

Comment:
I changed the variable names (really just `id`) to be a little more generic (I'm not sure if `encid` is a common term in the industry)

Reply:
Yeah I don't think it is. Good call!